artemis-protocols/artemis-openwire-protocol/src/main/java/org/apache/activemq/artemis/core/protocol/openwire/OpenWireConnection.java

```diff
@@ -779,11 +782,18 @@ private void doFail(ActiveMQException me, String message) {
       }
       try {
          if (this.getConnectionInfo() != null) {
-            protocolManager.removeConnection(this.getConnectionInfo(), me);
+            protocolManager.removeConnection(this.getConnectionInfo().getClientId(), this);
          }
-      } catch (InvalidClientIDException e) {
-         logger.warn("Couldn't close connection because invalid clientID", e);
       } finally {
+         try {
+            // doing the disconnect on the finally for extra caution.
+            // if removeConnection throws an exception of any kind, this should still happen
+            disconnect(false);
+         } catch (Throwable e) {
+            // it should never happen, but never say never
+            logger.warn("OpenWireConnection::disconnect failure", e);
```

Comment:
also here, I think, Throwable ignored and logger.debug. If this warn appears in a log, there is nothing a user can do. it will only be noise.